stdlib/public/Concurrency/TaskStatus.cpp

```diff
@@ -405,12 +405,21 @@ static void performCancellationAction(TaskStatusRecord *record) {
 
   // Child tasks need to be recursively cancelled.
   case TaskStatusRecordKind::ChildTask: {
+    fprintf(stderr, "[%s:%d] (%s) child task\n", __FILE__, __LINE__, __FUNCTION__);
     auto childRecord = cast<ChildTaskStatusRecord>(record);
     for (AsyncTask *child: childRecord->children())
       swift_task_cancel(child);
     return;
   }
 
+  case TaskStatusRecordKind::GroupChildTask: {
+    fprintf(stderr, "[%s:%d] (%s) group child task\n", __FILE__, __LINE__, __FUNCTION__);
+    auto childRecord = cast<GroupChildTaskStatusRecord>(record);
+    for (AsyncTask *child: childRecord->children())
+      swift_task_cancel(child);
+    return;
+  }
```

Comment:
Within `performCancellationAction` child tasks and group child tasks are effectively the same; the logic is the same the cast is different; perhaps can fold this into one...? Got the casts wrong when trying that before, but I can try harder